File: app/src/main/java/io/github/nexalloy/revanced/instagram/download/FeatureStatusTracker.java
package io.github.nexalloy.revanced.instagram.download;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Records which hooks actually resolved their target, so a failure to find an obfuscated
 * method is visible in the log instead of failing silently. InstaEclipse surfaced this in
 * its own UI; NexAlloy has no equivalent screen, so the state is only kept in memory and
 * logged once per hook.
 */
public final class FeatureStatusTracker {

    private static final Map<String, Boolean> HOOKED =
            Collections.synchronizedMap(new LinkedHashMap<>());

    private FeatureStatusTracker() {}

    public static void setHooked(String feature) {
        if (HOOKED.put(feature, Boolean.TRUE) == null) {
            ModuleLog.line("(NA|DL) hooked: " + feature);
        }
    }

    /** Kept for call-site compatibility; the string resource is only used by InstaEclipse's UI. */
    public static void setEnabled(String feature, int labelRes) {
        setHooked(feature);
    }

    public static boolean isHooked(String feature) {
        return Boolean.TRUE.equals(HOOKED.get(feature));
    }
}

File: app/src/main/java/io/github/nexalloy/revanced/instagram/download/FeatureStatusTracker.kt
package io.github.nexalloy.revanced.instagram.download

import java.util.Collections
import java.util.LinkedHashMap

/**
 * Records which hooks actually resolved their target, so a failure to find an obfuscated
 * method is visible in the log instead of failing silently. InstaEclipse surfaced this in
 * its own UI; NexAlloy has no equivalent screen, so the state is only kept in memory and
 * logged once per hook.
 */
object FeatureStatusTracker {

    private val HOOKED: MutableMap<String, Boolean> =
        Collections.synchronizedMap(LinkedHashMap())

    fun setHooked(feature: String) {
        if (HOOKED.put(feature, true) == null) {
            ModuleLog.line("(NA|DL) hooked: $feature")
        }
    }

    /** Kept for call-site compatibility; the string resource is only used by InstaEclipse's UI. */
    @Suppress("UNUSED_PARAMETER")
    fun setEnabled(feature: String, labelRes: Int) {
        setHooked(feature)
    }

    fun isHooked(feature: String): Boolean = HOOKED[feature] == true
}
